### ISSUE TITLE
Crash if distance to tree is near or beyond end of block

### ISSUE THREAD

**Issue by RickMohr** [azavea/nyc-trees#838]
Create a survey with one tree, whose distance is large (e.g. 1234), and submit.
500 Error:
```
line_interpolate_point: 2nd arg isnt within [0,1]
CONTEXT:  SQL function "st_line_interpolate_point" statement 2
PL/pgSQL function layoutboxes(geometry,boolean,double precision[],double precision[],double precision[],double precision[]) line 46 at assignment
SQL function "layoutboxes" statement 1
```

**Comment by dboyer** [comment]
Good catch. Moving to next sprint milestone to fix. Do we need info from Parks on max values?

**Comment by RickMohr** [comment]
I don't think we need info. A value measured on one block might be out of bounds on another block.

**Comment by dboyer** [comment]
True, especially since some of the avenues are very long blockfaces.

**Comment by steventlamb** [comment]
I did some research into this, and it's a little more complicated than it looks. At first I tried to limit the total distances to trees to the length of the blockface in a straight-forward manner, but there are 3 outstanding problems:
1. Parks does not want us to enforce this limit because slight user error and wheel error can make an otherwise well executed survey unusable.
2. This issue persists for surveys in which the total number of tree distances is *slightly less than* the total length of the blockface. (For example, blockface length of 232.25 ft, tree distances [100, 131].)
3. There's some noticeable scaling problems in the survey detail rendering that are probably related to this crash.

Here's a survey from south to north with four trees each spaced 50 feet apart, with a blockface length of 232.25 feet. The trees look evenly spaced, but the remaining distance does not look close to 32.25 feet.
![screenshot from 2015-03-30 15 15 29](https://cloud.githubusercontent.com/assets/1699455/6906036/1f6afb40-d6fa-11e4-8c39-1def352d1f8c.png)

I think we need to try to stop the crashing a different way, and see if we can avoid limited the survey distance sum entirely.

**Comment by hectcastro** [comment]
Just a note that we hit this ~5 times during user testing.

**Comment by dboyer** [comment]
Good to know. We need to talk to Parks about this.

**Comment by kdeloach** [comment]
@dboyer, we need to get going on this. Any progress?

**Comment by dboyer** [comment]
I'm setting up a call. More info shortly.

**Comment by dboyer** [comment]
See the fix in https://github.com/treekit/treekit/commit/3511f400873766a8c75912ca5dbb3500b1fc26a6

**Comment by dboyer** [comment]
The current code fix should clamp all trees to the end of the block edge if their distances would put them past the block edge. Data is not discarded.

**Comment by jwalgran** [comment]
We need to nail down some open source license details before pulling in any fixes, so we are blocked for now.

**Comment by kdeloach** [comment]
@jwalgran https://github.com/treekit/treekit/blob/master/COPYING

**Comment by kdeloach** [comment]
Previous comment withdrawn now that I understand the licensing issues a little better.

**Comment by dboyer** [comment]
Licensing details are confirmed. We can pull in the fix after #1612 is done.
